### DISCUSSION TITLE
Real Time Diarization Accuracy Improvement/Best Practices Help

### DISCUSSION THREAD

**Discussion by fahnub** [deepgram/community#439] [General help]
I am using real-time diarization and transcription in a web application. 

I am calling the following API to perform diarization:

> wss://api.deepgram.com/v1/listen?language=de&interim_results=false&diarize=true

I am experiencing some accuracy issues with the diarization functionality. 

For English, the recognition is sometimes inconsistent and does not recognize new speakers at all. 

For German, it does not detect new voices at all.

Are there any possible solutions to improve the accuracy of the diarization functionality? 

Any suggestions or recommendations are appreciated.

**Comment by jkroll-deepgram** [comment]
Hi @fahnub, I'm sorry to hear that diarization is proving to be difficult on your audio. Your API parameters look correct for streaming with diarization and a known language. Are you finding that the transcription itself is accurate, or is the transcription problematic as well? If you can provide any request IDs from requests where the diarization was poor, I can look into your audio and see if certain models or other features can help improve your results.

**Comment by catosaurusrex2003** [comment]
> I am using real-time diarization and transcription in a web application.
> 
> I am calling the following API to perform diarization:
> 
> > wss://api.deepgram.com/v1/listen?language=de&interim_results=false&diarize=true
> 
> I am experiencing some accuracy issues with the diarization functionality.
> 
> For English, the recognition is sometimes inconsistent and does not recognize new speakers at all.
> 
> For German, it does not detect new voices at all.
> 
> Are there any possible solutions to improve the accuracy of the diarization functionality?
> 
> Any suggestions or recommendations are appreciated.

hello there. 
can you please send the code snippet for how you are capturing audio and sending it to the endpoint using websockets.
